src/VideoStreaming/VideoReceiver.cc

```diff
@@ -196,30 +252,42 @@ void VideoReceiver::start()
             break;
         }
 
-        gst_bin_add_many(GST_BIN(_pipeline), dataSource, demux, parser, decoder, _videoSink, NULL);
+        if((_tee = gst_element_factory_make("tee", NULL)) == NULL)  {
+            qCritical() << "VideoReceiver::start() failed. Error with gst_element_factory_make('tee')";
+            break;
+        }
 
-        gboolean res = FALSE;
+        if((queue = gst_element_factory_make("queue", NULL)) == NULL)  {
+            qCritical() << "VideoReceiver::start() failed. Error with gst_element_factory_make('queue')";
+            break;
+        }
+
+        gst_bin_add_many(GST_BIN(_pipeline), dataSource, demux, parser, _tee, queue, decoder, _videoSink, NULL);
 
         if(isUdp) {
-            res = gst_element_link_many(dataSource, demux, parser, decoder, _videoSink, NULL);
+            // Link the pipeline in front of the tee
+            if(!gst_element_link_many(dataSource, demux, parser, _tee, queue, decoder, _videoSink, NULL)) {
+                qCritical() << "Unable to link elements.";
+                break;
+            }
         } else {
-            res = gst_element_link_many(demux, parser, decoder, _videoSink, NULL);
-        }
-
-        if (!res) {
-            qCritical() << "VideoReceiver::start() failed. Error with gst_element_link_many()";
-            break;
+            if(!gst_element_link_many(demux, parser, _tee, queue, decoder, _videoSink, NULL)) {
+                qCritical() << "Unable to link elements.";
+                break;
+            }
         }
 
-        dataSource = demux = parser = decoder = NULL;
+        dataSource = demux = parser = queue = decoder = NULL;
 
-        GstBus* bus = NULL;
+//        GstBus* bus = NULL;
 
-        if ((bus = gst_pipeline_get_bus(GST_PIPELINE(_pipeline))) != NULL) {
-            gst_bus_add_watch(bus, _onBusMessage, this);
-            gst_object_unref(bus);
-            bus = NULL;
-        }
+//        if ((bus = gst_pipeline_get_bus(GST_PIPELINE(_pipeline))) != NULL) {
+//            gst_bus_add_watch(bus, _onBusMessage, this);
+//            gst_object_unref(bus);
+//            bus = NULL;
+//        }
+        // Workaround for above watch on Windows
+        _busCheckTimer.start(0);
```

Comment:
See here: https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gstreamer/html/GstBus.html#gst-bus-enable-sync-message-emission 

You don't actually want to _become_ the bus sync handler, you just want to be signaled about the messages on the bus so you can act on them accordingly (such as EOS) without being responsible for handling them.